src/PageHeader/manageResponsiveValueWithCSS.ts

```diff
@@ -0,0 +1,130 @@
+import {isResponsiveValue, ResponsiveValue} from '../hooks/useResponsiveValue'
+import {BetterSystemStyleObject} from '../sx'
+
+function areAllValuesTheSame(responsiveValue: ResponsiveValue<boolean | number | string>): boolean {
+  if ('narrow' in responsiveValue && 'regular' in responsiveValue && 'wide' in responsiveValue) {
+    const values = Object.values(responsiveValue)
+    return values.every(value => value === values[0])
+  }
+  return false
+}
+function haveRegularAndWideSameValue(responsiveValue: ResponsiveValue<boolean | number | string>): boolean {
+  if ('regular' in responsiveValue && 'wide' in responsiveValue) {
+    return responsiveValue.regular === responsiveValue.wide
+  }
+  return false
+}
+
+/**
+ * This function is inspired by the `useResponsiveValue` hook and it's used to render responsive values with CSS.
+ * @param value - The value that needs to be rendered responsively
+ * @param cssProperty - The CSS property whoes value needs to be rendered responsively
+ * @param comparisonFunc - A function that evaluates which value to assign to the CSS property
+ 
+ * @example
+ * CSSManagedResponsiveValue({narrow: true, regular: true, wide: false}, 'display', value => {
+      return value ? 'none' : 'flex'
+    })
+ * @returns
+ * {
+ *   "@media screen and (max-width: 768px)": {
+ *     "display": "none"
+ *   },
+ *   "@media screen and (min-width: 768px)": {
+ *     "display": "none"
+ *   },
+ *   "@media screen and (min-width: 1440px)": {
+ *     "display": "flex"
+ *   }
+ * }
+ * 
+ * * @example
+ * CSSManagedResponsiveValue({regular: 'border.default', wide: 'canvas.inset'}, 'backgroundColor', (value): string => {
+    return value
+  })
+ * @returns
+ * {
+ *   "@media screen and (min-width: 768px)": {
+ *     "backgroundColor": "border.default"
+ *   },
+ *   "@media screen and (min-width: 1440px)": {
+ *     "backgroundColor": "canvas.inset"
+ *   }
+ * }
+ * 
+ * * @example
+* CSSManagedResponsiveValue({narrow: 'filled', regular: 'line'}, 'height', (value): string => {
+    return {
+      filled: 8,
+      line: 1,
+    }[value]
+  })
+ * @returns
+ * {
+ *  "@media screen and (max-width: 768px)": {
+ *     "height": 8
+ *   }
+ *   "@media screen and (min-width: 768px)": {
+ *     "height": 1
+ *   },
+ * }
+ */
+export function CSSManagedResponsiveValue<T>(
+  value: T,
+  cssProperty: string,
+  comparisonFunc: (value: T) => void,
+): BetterSystemStyleObject {
+  if (isResponsiveValue(value)) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const responsiveValue = value as Extract<T, ResponsiveValue<any>>
+
+    // Build media queries with the giving cssProperty and comparisonFunc
+    const narrowMediaQuery =
+      'narrow' in responsiveValue
+        ? {
+            [`@media screen and (max-width: 768px)`]: {
```

Comment:
Can you import these media query ranges from useResponsiveValue?

https://github.com/primer/react/blob/main/src/hooks/useResponsiveValue.ts#L9-L13


```js
import {viewportRanges} from '../utils/useResponsiveValue'

...

[`@media screen and ${viewportRanges.narrow}`]: ...
```

Reply:
Ah great idea!